### ISSUE TITLE
Upgrading Squeekboard will break keyboard size

### ISSUE THREAD

**Issue by wmww** [canonical/ubuntu-frame-osk#46]
We currently use Squeekboard v1.17.1, which works well. However v1.18.0, v1.19.0 and master at time of writing all draw the keyboard much too small, especially on nerrow, tall displays.

**Comment by Saviq** [comment]
@wmww with #53, I'm not seeing anything wrong?

![obraz](https://user-images.githubusercontent.com/231077/227965409-c801b66a-d941-4bc7-94e8-73d198f3dc42.png)

**Comment by AlanGriffiths** [comment]
Hmm, actually the above is a difference between `22/beta` and `22/edge`

**Comment by jrmcpeek** [comment]
#64 introduces this (unexpected? undesired?) behavior to the `22/beta` channel. It looks like the Canonical repos (https://git.launchpad.net/ubuntu/+source/squeekboard) point to 1.22, but the snaps are still flagged as 1.17:

```
22/beta:          65-sqbd1.17.1-0ubuntu1 2023-12-05 (525) 90MB -
```

This produces the documented issues above on a 1080p setup:

![image](https://github.com/MirServer/ubuntu-frame-osk/assets/55411348/a9803807-7ed1-4f2b-a1c5-9ff51fd0d30f)

**Comment by Saviq** [comment]
Thanks for noticing @jrmcpeek, I should've gone for `jammy` there.

https://packages.ubuntu.com/search?keywords=squeekboard

Will also fix the version string, that should've been taken from `debian/changelog` of the checked out source :/

**Comment by Saviq** [comment]
> Thanks for noticing @jrmcpeek, I should've gone for `jammy` there.

Oh, I _did_!

https://github.com/MirServer/ubuntu-frame-osk/blob/a5fc8b9b0afee324d485f8257ebd9cfcfed4c034/snap/snapcraft.yaml#L106-L108

So the version number is correct, it's just that the same problem occurred in the Jammy release.

It looks like one of these patches (which are backported in the Ubuntu source) must be the problem:

https://gitlab.gnome.org/World/Phosh/squeekboard/-/commit/68087a125cf26f364e200ef9f6aedeeb82218d66
https://gitlab.gnome.org/World/Phosh/squeekboard/-/merge_requests/548

Looks like we'll need to dig into it and find a proper fix.

**Comment by jrmcpeek** [comment]
@Saviq 

> Looks like we'll need to dig into it and find a proper fix.

So, until then, should pointing to the Ubuntu repos be reverted for beta and only tracked to edge?

As it stands, the usability of the `22/beta` is poor, but I'd also hate to move our pre-production devices to `22/stable` and then wake up one day and see that the production machines received an update exhibiting this same behavior without a chance to see it and raise the concern (as was done here).

**Comment by Saviq** [comment]
@jrmcpeek I reverted this change on `main` now, `beta` will get updated in a couple hours latest.

**Comment by svncibrahim** [comment]
I think I've fixed the sizing problem that started after Squeekboard v0.17.1. The main issue came from changes made to how the preferred_height was calculated, especially for the Librem 5's screen. To solve this, I created a patch using a system that scales based on a percentage of pixels, like how  [GNOME's virtual keyboard works](https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/ee03cf5f64ad7f14f8d61b32f07c831cc3bb6783/js/ui/keyboard.js#L1652)

I tested this patch on vertical and horizontal screens and didn't find any problems. I tested the patch using v1.24.0, but it should work for versions as far back as v19.0.0.

[height_scale.patch](https://github.com/canonical/ubuntu-frame-osk/files/14735877/height_scale.patch)

**Comment by Saviq** [comment]
Thanks @svncibrahim, that will help for sure, did you try submitting that upstream, yet?

**Comment by svncibrahim** [comment]
@Saviq I've sent it upstream. It's pending review now.

**Comment by Saviq** [comment]
Thanks @svncibrahim, where can I monitor it? I don't see it in https://gitlab.gnome.org/World/Phosh/squeekboard/-/merge_requests ?

**Comment by svncibrahim** [comment]
@Saviq I think we may have misunderstood each other. I had submitted this as a pull request to this repository. #72

**Comment by svncibrahim** [comment]
The scaling issue was resolved in version 1.42.0.

I tested the latest version, 1.43.1, and everything seems to be working fine now. I’ve attached two screenshots showing the before (v1.24.0) and after (v1.43.1), in 1080p.  #72 

v1.24.0 (without patch)
![1 24 0](https://github.com/user-attachments/assets/1a07df87-28ba-4999-b864-9c366a09f9c8)

v1.43.1
![1 43 1](https://github.com/user-attachments/assets/d7854272-0f6f-4b8d-bbe7-99b4a3d80105)
